### PR TITLE
Added @dbwiddis to OpenSearch maintainers

### PR DESCRIPTION
**Pull request by owaiskazi19**
### Description
I nominated Dan Widdis (@dbwiddis) to be a co-maintainer for Opensearch through our [nomination process](https://github.com/opensearch-project/.github/blob/main/RESPONSIBILITIES.md#becoming-a-maintainer). The maintainers have agreed, and Dan has kindly accepted.

Dan is already a maintainer of [opensearch-sdk-java](https://github.com/opensearch-project/opensearch-sdk-java) and working actively on Extensibility. Dan has authored several PRs on OpenSearch repo. Few of them got merged to `feature/extensions` branch and then later to `main`. 
1. [Add DynamicActionRegistry to ActionModule](https://github.com/opensearch-project/OpenSearch/pull/6734)
2. [Update ExtensionRestResponse to handle RestRequest object](https://github.com/opensearch-project/OpenSearch/pull/6874)
3. [Only store names of extension transport actions](https://github.com/opensearch-project/OpenSearch/pull/6545)
4. [Handle Deprecated and Replaced Routes in Extension Rest Handlers](https://github.com/opensearch-project/OpenSearch/pull/7064)
5. [getSettings extension point support](https://github.com/opensearch-project/OpenSearch/pull/4519)

Dan is active almost daily in the project, thoughtfully reviewing and commenting on pull requests ([example](https://github.com/opensearch-project/OpenSearch/pull/7302#discussion_r1178306885)). He has also closed [19 issues](https://github.com/opensearch-project/OpenSearch/issues?q=is%3Aissue+is%3Aclosed+assignee%3Adbwiddis) on core repo.

Few other contributions done by Dan:
1. [Avoid negative memory result in IndicesQueryCache stats calculation](https://github.com/opensearch-project/OpenSearch/pull/6917)
2. [Don't register destructiveDistroTest.docker twice](https://github.com/opensearch-project/OpenSearch/pull/6930)
3. [Remove deprecated org.gradle.util.DistributionLocator usage](https://github.com/opensearch-project/OpenSearch/pull/6212)
4. [Include Windows OS in Bootstrap initializeNatives() check for definitelyRunningAsRoot()](https://github.com/opensearch-project/OpenSearch/pull/4656)

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff
- [ ] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by ryanbogan — APPROVED** [review]

**Review by kotwanikunal — APPROVED** [review]

**Comment by nknize** [comment]
```
> Task :server:jarHell FAILED
| Output for C:\hostedtoolcache\windows\Java_Adopt_jdk\11.0.19-7\x64\bin\java.exe:#
# There is insufficient memory for the Java Runtime Environment to continue.
```

Sheesh! Brittle!  Can't even pass a maintainers update. 🤦🏻‍♂️ 

Re-firing

**Comment by nknize** [comment]
`DiskThresholdDeciderIT` strikes again... Love it when we're trolled by flakes.

```
./gradlew ':server:internalClusterTest' --tests "org.opensearch.cluster.routing.allocation.decider.DiskThresholdDeciderIT.testIndexCreateBlockIsRemovedWhenAnyNodesNotExceedHighWatermarkWithAutoReleaseEnabled" -Dtests.seed=62BA04CB1F0111FC -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=pl -Dtests.timezone=Indian/Chagos -Druntime.java=19
```

**Comment by dblock** [comment]
Do we need to add @dbwiddis to CODEOWNERS too? 

I adjusted permissions.

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -1 +1 @@
-*   @reta @anasalkouz @andrross @reta @Bukhtawar @CEHENKLE @dblock @gbbafna @setiah @kartg @kotwanikunal @mch2 @nknize @owaiskazi19 @Rishikesh1159 @ryanbogan @saratvemulapalli @shwetathareja @dreamer-89 @tlfeng @VachaShah @xuezhou25
+*   @reta @anasalkouz @andrross @reta @Bukhtawar @CEHENKLE @dblock @gbbafna @setiah @kartg @kotwanikunal @mch2 @nknize @owaiskazi19 @Rishikesh1159 @ryanbogan @saratvemulapalli @shwetathareja @dreamer-89 @tlfeng @VachaShah @xuezhou25 @dbwiddis
```
Comment (owaiskazi19) [inline .github/CODEOWNERS:1]: Not sure why this error for @xuezhou25 here?

**Review by dblock — APPROVED** [review]
Everything looks good. 

@xuezhou25 please speak reach out if you want your maintainer permissions restored - that was automatically adjusted when leaving the org

File: .github/CODEOWNERS (hunk 1 above)
Comment (dblock) [reply .github/CODEOWNERS]: Account was automatically removed from the opensearch-project org. That's what currently happens when you work for Amazon - @hyandell help us fix that?

**Review by anasalkouz — APPROVED** [review]

**Pull request merged**